### PR TITLE
ADEN-4628 Veles bidder

### PR DESCRIPTION
**Pull request by fraszczakszymon**
Our own prebid adapter for DFP video units

### CODE REVIEW

File: extensions/wikia/AdEngine/js/lookup/prebid/adapters/veles.js [hunk 1]
```diff
@@ -0,0 +1,165 @@
...
+		if (adParameters.childNodes.length) {
+			pairs = adParameters.childNodes[0].nodeValue.split(',');
+		}
+		pairs.forEach(function (pair) {
+			var data = pair.split('=');
+
+			parameters[data[0]] = data[1];
+		});
+
+		return parameters;
```
Comment (dianafa) [inline extensions/wikia/AdEngine/js/lookup/prebid/adapters/veles.js]: parsing query string looks like more generic functionality, what about moving it to some utils?

File: extensions/wikia/AdEngine/js/lookup/prebid/adapters/veles.js [hunk 2]
```diff
@@ -0,0 +1,165 @@
...
+			addBids(bidderRequest, vastRequest.response, price);
+		} else {
+			addEmptyBids(bidderRequest);
+		}
+	}
+
+	function requestVast(bidderRequest) {
+		var request = new win.XMLHttpRequest(),
+			skin = adContext.getContext().targeting.skin,
+			vastUrl = vastUrlBuilder.build(640 / 480, {
```
Comment (dianafa) [inline extensions/wikia/AdEngine/js/lookup/prebid/adapters/veles.js:135]: will it always be 640 / 480?

File: extensions/wikia/AdEngine/js/lookup/prebid/adapters/veles.js [hunk 3]
```diff
@@ -0,0 +1,165 @@
...
+			parameters = {};
+
+		if (adParameters.childNodes.length) {
+			pairs = adParameters.childNodes[0].nodeValue.split(',');
+		}
+		pairs.forEach(function (pair) {
+			var data = pair.split('=');
+
+			parameters[data[0]] = data[1];
+		});
```
Comment (dianafa) [inline extensions/wikia/AdEngine/js/lookup/prebid/adapters/veles.js]: what about
```
var parameters = {};

if (adParameters.childNodes.length) {
  adParameters.childNodes[0].nodeValue.split(',').forEach(function (pair) {
    var data = pair.split('=');
    parameters[data[0]] = data[1];
});
}
```

File: extensions/wikia/AdEngine/js/lookup/prebid/adapters/veles.js (hunk 1 above)
Comment (fraszczakszymon) [reply extensions/wikia/AdEngine/js/lookup/prebid/adapters/veles.js]: It's not query string. Example:
https://github.com/Wikia/app/pull/12351/files/9fd9af8784265e4d6f4069821114eee599d0c972#diff-b78f0cb174bc71f686b176179584f4d9R82
What about extracting it when it's needed for the second time?

File: extensions/wikia/AdEngine/js/lookup/prebid/adapters/veles.js (hunk 2 above)
Comment (fraszczakszymon) [reply extensions/wikia/AdEngine/js/lookup/prebid/adapters/veles.js:135]: Currently we are using 640x480 only

File: extensions/wikia/AdEngine/js/lookup/prebid/adapters/veles.js [hunk 4]
```diff
@@ -0,0 +1,163 @@
...
+
+		if (adParameters.childNodes.length && adParameters.childNodes[0].nodeValue) {
+			adParameters.childNodes[0].nodeValue.split(',').forEach(function (pair) {
+				var data = pair.split('=');
+
+				parameters[data[0]] = data[1];
+			});
+		}
+
+		return parameters;
```
Comment (nandy-andy) [inline extensions/wikia/AdEngine/js/lookup/prebid/adapters/veles.js:47]: Are you sure `parameters` won't be always `{}`? My memory isn't always perfect but I wonder if `forEach` isn't asynchronous.

File: extensions/wikia/AdEngine/js/lookup/prebid/adaptersRegistry.js [hunk 5]
```diff
@@ -1,15 +1,23 @@
+/*global define*/
 define('ext.wikia.adEngine.lookup.prebid.adaptersRegistry', [
 	'ext.wikia.adEngine.lookup.prebid.adapters.aol',
 	'ext.wikia.adEngine.lookup.prebid.adapters.appnexus',
-	'ext.wikia.adEngine.lookup.prebid.adapters.indexExchange'
-], function(aol, appnexus, index) {
+	'ext.wikia.adEngine.lookup.prebid.adapters.indexExchange',
+	'ext.wikia.adEngine.lookup.prebid.adapters.wikia',
```
Comment (nandy-andy) [inline extensions/wikia/AdEngine/js/lookup/prebid/adaptersRegistry.js:6]: What's Wikia adapter? Do we want to add it here?

File: extensions/wikia/AdEngine/js/lookup/prebid/adapters/veles.js (hunk 4 above)
Comment (fraszczakszymon) [reply extensions/wikia/AdEngine/js/lookup/prebid/adapters/veles.js:47]: Discussed offline. Looks good

File: extensions/wikia/AdEngine/js/lookup/prebid/adaptersRegistry.js (hunk 5 above)
Comment (fraszczakszymon) [reply extensions/wikia/AdEngine/js/lookup/prebid/adaptersRegistry.js:6]: Discussed offline. Looks good

**Comment by bognix** [comment]
LGTM

**Pull request merged**